WebPhapp/backend/mysql_helper.js

```diff
@@ -35,5 +35,102 @@ module.exports = {
                 resolve({rows, fields});
             });
         });
-    }
+    },
+
+    /*
+    Insert a user into the DB
+    Args:
+        username: Username to insert
+        password: The salted and hashed password
+        role: The type of user being inserted
+        connection: MySQL Connection object
+    Returns: Promise.
+        Upon resolution, returns the insertion ID of the row.
+    */
+    insertUser: function(username, password, role, connection) {
+        var q = `
+        INSERT INTO users (role, username, password)
+        VALUES (?,?,?); SELECT LAST_INSERT_ID();
+        `;
+
+        return new Promise((resolve, reject) => {
+            var values = [role.toString(),username,password];
+            connection.query(q, values, (error, rows, fields) => {
+                if (error) reject(error);
+                resolve({rows, fields});
+            });
+        });
+    },
+
+
+    /*
+    Insert the salt into the DB
+    Args:
+        userID: userID to insert
+        salt: Salt value for the user
+        connection: MySQL Connection object
+    Returns: Promise.
+        Nothing is inside of the promise.
+    */
+    insertSalt: function(userID, salt, connection){
+        var q = `
+        INSERT INTO salts
+        VALUES (?,?);
+        `;
+
+        return new Promise((resolve, reject) => {
+            var values = [userID,salt];
+            connection.query(q, values, (error, rows, fields) => {
+                if (error) reject(error);
+                resolve({rows, fields});
+            });
+        });
+    },
+
+    /*
+    Get the value of a salt, given the username
+    Args:
+        username: the username of the user
+        connection: MySQL Connection object
+    Returns: Promise
+        Upon resolution the salt of a given user.
+    */
+    getSaltByUsername: function(username, connection){
+        var q = `
+        SELECT salt
+        FROM salts s, users u
+        WHERE s.id = u.id AND u.username = ? LIMIT 1;
```

Comment:
Convention I am familiar with puts LIMIT on a new line. Not a big deal.

Reply:
Fixed this and several other SQL styling issues.